google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/SessionPoolTest.java

```diff
@@ -564,6 +579,89 @@ public void keepAlive() throws Exception {
     pool.closeAsync().get();
   }
 
+  @Test
+  public void blockAndTimeoutOnPoolExhaustion() throws InterruptedException, ExecutionException {
+    if (minSessions != 0) {
+      // Only execute for minSessions == 0 as we need to setup and shutdown a mock sampler during
+      // the test case. A second run would not work as there is no way to restart the sampler.
+      return;
+    }
+    // Setup a dummy trace handler.
+    final AtomicInteger deadlineExceededCount = new AtomicInteger();
+    Handler handler =
+        new Handler() {
+          @Override
+          public void export(Collection<SpanData> spanDataList) {
+            for (SpanData sd : spanDataList) {
+              if (sd.getStatus() == Status.DEADLINE_EXCEEDED) {
```

Comment:
Should this counter only be incremented for spans that match WAIT_FOR_SESSION?

Reply:
Agree, that makes the test case less prone to false failures.